### PR TITLE
Added fallback format to parse offsets that don't have colon and addi…

### PR DESCRIPTION
**Pull request by rma-rripken**
…tional test cases for format that was given in Swagger doc and also example from format TimeSeries uses.

### CODE REVIEW

**Comment by rma-rripken** [comment]
This is related to #147 and #74

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java [hunk 1]
```diff
@@ -1,58 +1,174 @@
...
+            + "(:?\\d{2})?"  // optional colon and 2 digit minute offset
+            + "|Z" // OR Z
+            + "|\\[" // OR an open square bracket
+            + "[a-zA-Z0-9]*" // any number of letters and numbers
+            + "(/[a-zA-Z_]*)?" // optionally followed by a / and more letters or underbar as in Los_Angeles
+            + "]" // followed by a close square bracket
+            + ")"
+            + "$" // then String must immediately end
+            );
+
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java:38]: I don't think I've ever seen anyone layout a regular expression like this... thank you.

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java [hunk 2]
```diff
@@ -1,58 +1,174 @@
...
+     * @return The parsed ZonedDateTime
+     */
+    public static ZonedDateTime parseUserDate(String text, String timezone) {
         ZoneId tz = ZoneId.of(timezone);
-        return parseUserDate(date, tz, ZonedDateTime.now(tz));
+        return parseUserDate(text, tz, ZonedDateTime.now(tz));
     }
 
-    public static ZonedDateTime parseUserDate(String date, ZoneId tz, ZonedDateTime now){
+    public static ZonedDateTime parseUserDate(String date, ZoneId tz, ZonedDateTime now) {
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java]: date here should likely also be "text" to be consistent.

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java [hunk 3]
```diff
@@ -1,58 +1,174 @@
...
+                // Still couldn't parse so might as well throw the original exception.
+                throw e;
+            }
+        }
+
+        return zdt;
     }
 
-    private static ZonedDateTime parserUserPeriod(String date, ZoneId tz, ZonedDateTime now) {
+    private static ZonedDateTime firstMatch(String text, String[] possibleDateFormats) {
```
Comment (MikeNeilson) [inline cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java]: nitpicky, but maybe return optional instead of null?

**Review by MikeNeilson — DISMISSED** [review]

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java (hunk 2 above)
Comment (rma-rripken) [reply cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java]: agreed

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java (hunk 3 above)
Comment (rma-rripken) [reply cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java]: Agreed, I forget about Optional.

File: cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java (hunk 1 above)
Comment (rma-rripken) [reply cwms_radar_api/src/main/java/cwms/radar/helpers/DateUtils.java:38]: Regex, in general, is missing ergonomics.  I ended up picking apart the regex just to understand it so I figured I'd leave the comments for future readers.  I know there are some Regex builder libraries but we don't ever seem to do enough regex to justify bringing one in.

**Review by MikeNeilson — APPROVED** [review]

**Comment by rma-rripken** [comment]
This is a significant change for user input that included zone/offset information.   I think the old behavior was wrong. I encourage reviewers to double check this PR b/c I really don't want to get this wrong.

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**